rimage/transform/parallel_projection_test.go

```diff
@@ -0,0 +1,135 @@
+package transform
+
+import (
+	"fmt"
+	"image"
+	"os"
+	"testing"
+
+	"github.com/golang/geo/r3"
+	"go.viam.com/test"
+	"go.viam.com/utils/artifact"
+
+	pc "go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/rimage"
+	"go.viam.com/rdk/spatialmath"
+)
+
+// Helper function that makes and returns a PointCloud from an artifact path consisting of the first numPoints of points.
+func makePointCloudFromArtifact(t *testing.T, artifactPath string, numPoints int) (pc.PointCloud, error) {
+	t.Helper()
+	pcdFile, err := os.Open(artifact.MustPath(artifactPath))
+	if err != nil {
+		return nil, err
+	}
+	pcd, err := pc.ReadPCD(pcdFile)
+	if err != nil {
+		return nil, err
+	}
+
+	if numPoints == 0 {
+		return pcd, nil
+	}
+
+	shortenedPC := pc.NewWithPrealloc(numPoints)
+
+	counter := numPoints
+	pcd.Iterate(0, 0, func(p r3.Vector, d pc.Data) bool {
+		if counter > 0 {
+			err = shortenedPC.Set(p, d)
+			counter--
+		}
+		return err == nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return shortenedPC, nil
+}
+
+func TestParallelProjectionOntoXZWithRobotMarker(t *testing.T) {
+	t.Run("Project an empty pointcloud", func(t *testing.T) {
+		p := spatialmath.NewPose(r3.Vector{X: 0, Y: 0, Z: 0}, spatialmath.NewOrientationVector())
+		ppRM := NewParallelProjectionOntoXZWithRobotMarker(&p)
+
+		pointcloud := pc.New()
+
+		im, unusedDepthMap, err := ppRM.PointCloudToRGBD(pointcloud)
+		test.That(t, err.Error(), test.ShouldContainSubstring, "calculation of the X-coord's mean during pcd projection failed")
+		test.That(t, im, test.ShouldBeNil)
+		test.That(t, unusedDepthMap, test.ShouldBeNil)
+	})
+
+	t.Run("Project a single point pointcloud with no data", func(t *testing.T) {
```

Comment:
Can this test check that the point and robot marker are set correctly in the image?

Reply:
+1 for this.

Also, in this case, since the single point is at [0, 0, 0], this test would not have caught the case we discussed above, where points might wrongly all be located at [0, 0] in the picture due to the error in scaling. 

[q] Can we use a point that's not at [0, 0, 0]? And consider doing the same in the tests below?

Reply:
Checks made in two of the three test cases. One with point at origin and one with 2 points (+ robot marker) for a non-zero defining corner point